### PR TITLE
update coredns to v1.11.1

### PR DESCRIPTION
**Pull request by tukwila**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:

- [x] https://github.com/coredns/corefile-migration/pull/76
- [x] https://github.com/kubernetes/k8s.io/pull/5701
- [x] https://github.com/coredns/coredns/releases/tag/v1.11.0

- [x] https://github.com/coredns/corefile-migration/pull/78
- [x] https://github.com/kubernetes/k8s.io/pull/5743
- [x] https://github.com/coredns/coredns/releases/tag/v1.11.1

#### Which issue(s) this PR fixes:

Fixes # None

#### Does this PR introduce a user-facing change?

```release-note
upgrade coredns to v1.11.1
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by tukwila** [comment]
@dims @chrisohaver could please review this PR? big thanks.

**Comment by chrisohaver** [comment]
IMO, we should not use 1.11.0 due to the performance regression it introduces. It is fixed in 1.11.1.

**Review by neolit123 — COMMENTED** [review]
/lgtm
/approve
for kubeadm

**Comment by neolit123** [comment]
> IMO, we should not use 1.11.0 due to the performance regression it introduces. It is fixed in 1.11.1.

+1

**Comment by tukwila** [comment]
> > IMO, we should not use 1.11.0 due to the performance regression it introduces. It is fixed in 1.11.1.
> 
> +1 @tukwila can you please help update to .1?

yes, https://github.com/coredns/corefile-migration/pull/78

**Review by neolit123 — COMMENTED** [review]
thanks
/lgtm
/ok-to-test

File: build/dependencies.yaml [hunk 1]
```diff
@@ -39,7 +39,7 @@ dependencies:
       match: registry.k8s.io/coredns
 
   - name: "coredns-kubeadm"
-    version: 1.10.1
+    version: 1.11.0
```
Comment (pacoxu) [inline build/dependencies.yaml:42]: 1.11.1

Comment (tukwila) [reply build/dependencies.yaml]: corrected.

**Comment by neolit123** [comment]
please check the failures here (уеllow text)
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/120116/pull-kubernetes-unit/1694877880716103680

i think this means the coredns migration library also needs an update.
github.com/coredns/corefile-migration

perhaps the latest migration lib supports upgrade to 1.11.1

**Comment by tukwila** [comment]
> please check the failures here (уеllow text) https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/120116/pull-kubernetes-unit/1694877880716103680
> 
> i think this means the coredns migration library also needs an update. github.com/coredns/corefile-migration
> 
> i.e. perhaps the latest migration lib supports upgrade to 1.11.1

yes, let me consult with  @chrisohaver .

**Review by neolit123 — COMMENTED** [review]
/assign @dims 
ptal for dep / root approve

**Comment by neolit123** [comment]
please see the errors here
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/120116/pull-kubernetes-dependencies/1695090910880075776

you need to run ./hack/update-vendor.sh locally.

also check the other tests for failures.

**Comment by pacoxu** [comment]
/lgtm
/unhold
ping @dims for approva

**Comment by tukwila** [comment]
@dims could you please look at this pr?

**Comment by dims** [comment]
i see 2 `/lgtm`(s) from folks who know this stuff :) looks good to me as well.

/approve
/lgtm

**Pull request merged**